File: trunk/ShadowFrameworkLite20/src/shadow/math/SFValue1f.java
package shadow.math;

public class SFValue1f extends SFValuenf{
	
	/*@note package visibility*/
	SFValue1f(int n) {
		super(n);
	}
	
	public SFValue1f(float x) {
		super(1);
		this.v[0] = x;
	}
	
	public SFValue1f(double x) {
		super(1);
		this.v[0] = (float)x;
	}
	
	public void storeOn1f(float []f){
		f[0]=v[0];
	}
	
	public float getX(){
		return v[0];
	}
	
	public void setX(float x){
		v[0]=x;
	}

	public void mult1f(float m){
		v[0]*=m;
	}
	
	public void add1f(float dX){
		v[0]+=dX;
	}
	
	public void subtract1f(float dX){
		v[0]+=dX;
	}
	
	public void setByIndex(int index,float val){
		if(index==0)
			v[0]=val;
	}

	@Override
	public SFValuenf cloneValue() {
		return new SFValue1f(this.v[0]);
	}
	
}

File: trunk/ShadowFrameworkJS/src/shadow/math/SFValue1f.js
//Java to JS on 06/02/2012

function SFValue1f(x){
	this.v=new Array();
	this.v[0]=x;
}

inherit(SFValue1f,SFValuenf);
	
		
SFValue1f.prototype["storeOn1f"]=function(f){
			f[0]=this.v[0];
		};
		
SFValue1f.prototype["getX"]=function(){
		return this.v[0];
		};
		
SFValue1f.prototype["setX"]=function(x){
			this.v[0]  = x;
		};
		
SFValue1f.prototype["mult1f"]=function(m){
			this.v[0] *= m;
		};
		
SFValue1f.prototype["add1f"]=function(dX){
			this.v[0] += dX;
		};
		
SFValue1f.prototype["subtract1f"]=function(dX){
			this.v[0] -= dX;
		};
		
SFValue1f.prototype["setByIndex"]=function(index, val){
			this.v[index]=val;
		};
		
SFValue1f.prototype["getByIndex"]=function(index){
			if(index==0)
				return this.v[index];
		};
		
SFValue1f.prototype["cloneValue"]=function(){
			return new SFValue1f(this.v[0]);
		};
